sound/soc/sof/intel/cnl.c

```diff
@@ -156,6 +158,23 @@ static int cnl_ipc_is_ready(struct snd_sof_dev *sdev)
 	return 1;
 }
 
+static int cnl_ipc_reply_done(struct snd_sof_dev *sdev)
+{
+	/* clear DONE bit - tell DSP we have completed the operation */
+	snd_sof_dsp_update_bits_forced(sdev, HDA_DSP_BAR,
+				       CNL_DSP_REG_HIPCIDA,
+				       CNL_DSP_REG_HIPCIDA_DONE,
+				       CNL_DSP_REG_HIPCIDA_DONE);
+
+	/* unmask Done interrupt */
+	snd_sof_dsp_update_bits(sdev, HDA_DSP_BAR,
+				CNL_DSP_REG_HIPCCTL,
+				CNL_DSP_REG_HIPCCTL_DONE,
+				CNL_DSP_REG_HIPCCTL_DONE);
+
+	return 0;
```

Comment:
This looks like the same flow in the irq thread.  If so, IRQ thread should call this directly.